apps/distgo/pkg/script/script.go

```diff
@@ -92,3 +94,20 @@ func WriteAndExecute(buildSpec params.ProductBuildSpec, script string, stdOut, s
 	}
 	return nil
 }
+
+func GetBuildArgs(buildSpec params.ProductBuildSpec, script string) ([]string, error) {
+	stdoutBuf := bytes.Buffer{}
+	stderrBuf := bytes.Buffer{}
+	combinedBuf := bytes.Buffer{}
+	stdoutMW := io.MultiWriter(&stdoutBuf, &combinedBuf)
+	stderrMW := io.MultiWriter(&stderrBuf, &combinedBuf)
+	if err := WriteAndExecute(buildSpec, script, stdoutMW, stderrMW, nil); err != nil || stderrBuf.String() != "" {
+		return nil, errors.Wrapf(err, "failed to execute build args script for %v: %v", buildSpec.ProductName, combinedBuf.String())
+	}
+
+	buildArgsString := strings.TrimSpace(stdoutBuf.String())
+	if buildArgsString != "" {
+		return strings.Split(buildArgsString, "\n"), nil
+	}
+	return []string{}, nil
```

Comment:
This comment still stands -- should flip so that, if `buildArgsString == ""`, then we return `nil, nil`, and in the other case we do this operation.